src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1926,4 +1942,27 @@ public static String getVersionName() {
         // value
         return "1.1.12";
     }
+
+    /**
+     * A {@link Future}  whose result can be set by a {@link #set(Object)} or {@link #setException(Throwable)}
+     */
+    final static class SettableFuture<V> extends FutureTask<V> {
```

Comment:
This approach will cause break to Sync method, I'll keep the strategy with SettableFuture, but I create a method that can avoid all this hacky staff. And marked old method as deprecated, when clients will switch we can remove old method with SettableFuture